src/Symfony/Component/Yaml/Parser.php

```diff
@@ -381,6 +381,11 @@ private function doParse(string $value, int $flags)
                         if (0 === $this->offset && !$deprecatedUsage && isset($line[0]) && ' ' === $line[0]) {
                             throw new ParseException('Unable to parse.', $this->getRealCurrentLineNb() + 1, $this->currentLine, $this->filename);
                         }
+
+                        if ($this->offset >= 1 && preg_match('/:/', $this->currentLine) && $this->isNextLineIndented()) {
```

Comment:
we should use `strpos()` instead of `preg_match()` here

Reply:
```yaml
data:
    dbal:wrong
        default_connection:monolith
```

Would trigger the deprecation too, wouldn't it?

Reply:
IMO that's not right. The first example is valid YAML. But I guess we could fix that by changing this to `false !== strpos($this->currentLine, ': ')`.

Reply:
How can we check that ? Looking at the YAML 1.2 specification, I see 
"
Each node must be indented further than its parent node. All sibling nodes must use the exact same indentation level. However the content of each sibling node may be further indented independently.
"

Reply:
What do you mean with what to check? I am talking here about the colon not being followed by a space.

Reply:
Are you sure the first example is also a valid YAML? Quoting from the specification : 
"Normally, YAML insists the “:” mapping value indicator be separated from the value by white space."
IMO it should also trigger the deprecation notice.

Reply:
I replaced preg_match with strpos.

Reply:
Yes, in that case the last two lines are not mappings, but form a string that crosses multiple lines. This string in turn is the value of the `data` mapping key.

Reply:
Ok I updated the check in order to trigger deprecation notice only on second scenario. I also updated the multiline test to make sure the value will still be correctly generated with a colon in the string.